pulsar-common/src/main/java/org/apache/pulsar/common/naming/Metadata.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.common.naming;
+
+import java.util.Map;
+
+public class Metadata {
+
+    private static final int MAX_KEY_VALUE_PAIRS = 8;
```

Comment:
Maybe we could have just 1 single limit. Let say 1Kb, and we could just check the length of all keys and values are under that limit. That would be more flexible.

Reply:
Updated to allow for 1 Kb.